### PR TITLE
RFC: Fix 2:1 spacing

### PR DESCRIPTION
**Pull request by krkshitij**

### CODE REVIEW

File: docs/rfcs/fix-2-to-1-spacing.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+![](../assets/images/2-to-1-spacing-problem.png)
+
+- The bars don’t occupy the entire space due to the default bar width of 16px and a fixed 2:1 spacing, often leaving considerable amount of blank space before the first bar and after the last bar. This becomes particularly noticeable when the chart width is large.
+- Additionally, this setup causes unnecessary overlapping of x-axis labels.
+
+## Proposal
+
+1. Add optional `xAxisInnerPadding`, `xAxisOuterPadding` props to vertical bar charts.
+   - `xAxisInnerPadding` controls the space between bars, while `xAxisOuterPadding` controls the space before the first bar and after the last bar.
+   - These props accept values between 0 and 1, representing a fraction of the [step](https://d3js.org/d3-scale/band#band_step), which is the interval between the start of a bar and the start of the next bar. These props are particularly relevant when using a string x-axis. For additional information on padding in string axes, see [Band scales | D3 by Observable](https://d3js.org/d3-scale/band#band_paddingInner).
```
Comment (AtishayMsft) [inline docs/rfcs/fix-2-to-1-spacing.md:22]: Add a section for definitions. Define step there. It is little confusing here.

File: docs/rfcs/fix-2-to-1-spacing.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+   - By default, the inner padding is set to 2/3, meaning the space between bars is twice the bar width, while the outer padding is set to 1/3, implying that the space before the first bar and after the last bar is equivalent to the bar width. These default values are calculated using the formula:
+     $$
+     innerPadding = {spaceBetweenBars \over spaceBetweenBars + barWidth}
+     $$
+     For a more detailed explanation of how these values were derived, see [Implementing 2:1 spacing | FluentUI Charting Contrib Docsite](https://microsoft.github.io/fluentui-charting-contrib/docs/implementing-2-to-1-spacing).
```
Comment (AtishayMsft) [inline docs/rfcs/fix-2-to-1-spacing.md:27]: keep this path as relative

File: docs/rfcs/fix-2-to-1-spacing.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+   - The default outer padding is non-zero, but users have the option to override it with a value of 0 if they prefer no space before the first bar and after the last bar. Therefore, the minimum domain margin of 8px specified in the design is disregarded for string x-axis.
+2. Use the existing `barWidth` prop and add an optional `maxBarWidth` prop to vertical bar charts.
+   - The `barWidth` prop takes precedence over the props mentioned above. If not explicitly specified, the bar width will be dynamically adjusted to avoid overlapping while maintaining specified spacing between and around the bars. However, when `barWidth` is provided, it will be respected, potentially compromising padding.
+   - In cases where the chart width is large, the bar width might become excessively large. To mitigate this, the `maxBarWidth` prop can be utilized to restrict the width of the bars to a specified maximum.
```
Comment (AtishayMsft) [inline docs/rfcs/fix-2-to-1-spacing.md:31]: Create a decision tree covering the whole logic. Start with - axis type -> numeric or data -> no action. axis type -> string -> follow the order as you have described.

File: docs/rfcs/fix-2-to-1-spacing.md (hunk 1 above)
Comment (krkshitij) [reply docs/rfcs/fix-2-to-1-spacing.md]: done

File: docs/rfcs/fix-2-to-1-spacing.md (hunk 2 above)
Comment (krkshitij) [reply docs/rfcs/fix-2-to-1-spacing.md]: done

File: docs/rfcs/fix-2-to-1-spacing.md (hunk 3 above)
Comment (krkshitij) [reply docs/rfcs/fix-2-to-1-spacing.md:33]: done

File: apps/docsite/sidebars.ts [hunk 4]
```diff
@@ -58,6 +58,16 @@ const sidebars: SidebarsConfig = {
...
+    {
+      type:'category',
+      label: 'RFCs',
+      items: [
+        {
+          type: 'autogenerated',
+          dirName: 'rfcs'
+        }
+      ]
+    },
```
Comment (AtishayMsft) [inline apps/docsite/sidebars.ts:70]: I guess the spacing fix flow png is not needed now

Comment (krkshitij) [reply apps/docsite/sidebars.ts:70]: removed it

File: docs/rfcs/fix-2-to-1-spacing.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+## Problem
+
+![](../assets/images/2-to-1-spacing-problem.png)
+
+- The bars don’t occupy the entire space due to the default bar width of 16px and a fixed 2:1 spacing, often leaving considerable amount of blank space before the first bar and after the last bar. This becomes particularly noticeable when the chart width is large.
+- Additionally, this setup causes unnecessary overlapping of x-axis labels.
+
+## Proposal
+
+<iframe allowfullscreen frameborder="0" width="640" height="480" src="https://lucid.app/documents/embedded/86f214c8-0d6d-4640-a81a-fa492851f6d9" id="iv-eZ8WS-fIH"></iframe>
```
Comment (AtishayMsft) [inline docs/rfcs/fix-2-to-1-spacing.md:20]: Iframe with cross domain content is not a good security practice. It is not rendering in the browser also.

Comment (krkshitij) [reply docs/rfcs/fix-2-to-1-spacing.md]: Replaced it with image linked to diagram file

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**